packages/engine/lib/memory/src/shared_memory/shared_memory.rs

```diff
@@ -13,18 +13,63 @@ use crate::{
     },
 };
 
-pub type Buffers<'a> = (&'a [u8], &'a [u8], &'a [u8], &'a [u8]);
+/// An identifier for a shared memory section.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct MemoryId<'id> {
+    id: &'id Uuid,
+    suffix: u16,
+}
+
+impl<'id> MemoryId<'id> {
+    /// Creates a new identifier from the provided [`Uuid`].
+    ///
+    /// This will generate a suffix and ensures, that the shared memory segment does not already
+    /// exists at */dev/shm/*.
+    pub fn new(id: &'id Uuid) -> Self {
+        loop {
+            let memory_id = Self {
```

Comment:
Every time we are generating a new suffix, it would require to loop again. Or we need a global variable tracking the highest number used. I expect this to be much slower, as I expect `Path::exists()` to be slow.

Reply:
> as I expect Path::exists() to be slow

I'm not as confident in that as you (although definitely ridiculously slow in real terms compared to simply generating random numbers), but I also don't think we actually create new id's that frequently. (Meaning I also don't expect the loop bounds to be particularly large, and it's possible that we'd be reusing numbers as we release memory too, which would decrease the loop iterations by even more)

Anyway I think you bring up a good point and I wouldn't want to change this without some benchmarking in place. Going to resolve the comment.